image_builder/audiomini/build_audiomini_image.sh

```diff
@@ -0,0 +1,87 @@
+
+# Copy DistroBlueprint.xml into LinuxBootImageFileGenerator
+cp DistroBlueprint.xml LinuxBootImageFileGenerator/;
+
+# Make the Image_partitions/Pat_1_vfat Folder
+mkdir -p LinuxBootImageFileGenerator/Image_partitions/Pat_1_vfat/;
+
+# Check for the Audio Mini DTB
+if [ -f soc_system.dtb ]; then
+    echo 'Using the local Audio Mini DTB.'
+else
+    echo 'Downloading the Audio Mini DTB from S3 Archive'
+    wget https://frost-release.s3-us-west-2.amazonaws.com/linux-images/artifacts/audiomini/soc_system.dtb
+fi
+cp soc_system.dtb LinuxBootImageFileGenerator/Image_partitions/Pat_1_vfat/;
+
+# Check for the Audio Mini Raw Binary File
+if [ -f soc_system.rbf ]; then
+    echo 'Using the local Audio Mini RBF.'
+else
+    echo 'Downloading the Audio Mini RBF from S3 Archive'
+    wget https://frost-release.s3-us-west-2.amazonaws.com/linux-images/artifacts/audiomini/soc_system.rbf
+fi
+cp soc_system.rbf LinuxBootImageFileGenerator/Image_partitions/Pat_1_vfat/;
+
+# Check for the zImage
+if [ -f zImage ]; then
+    echo 'Using the local zImage.'
+else
+    echo 'Downloading the zImage from S3 Archive'
+    wget https://frost-release.s3-us-west-2.amazonaws.com/linux-images/artifacts/common/zImage
+fi
+cp zImage LinuxBootImageFileGenerator/Image_partitions/Pat_1_vfat/;
+
+# Check for the Audio Mini uBoot Source
+if [ -f u-boot.scr ]; then
+    echo 'Using the local Audio Mini uBoot Source.'
+else
+    echo 'Downloading the Audio Mini uBoot Source from S3 Archive'
+    wget https://frost-release.s3-us-west-2.amazonaws.com/linux-images/artifacts/audiomini/u-boot.scr
+fi
+cp u-boot.scr LinuxBootImageFileGenerator/Image_partitions/Pat_1_vfat/;
+
+# Check for the Audio Mini uBoot Image
+if [ -f u-boot.img ]; then
+    echo 'Using the local Audio Mini uBoot Image.'
+else
+    echo 'Downloading the Audio Mini uBoot Image from S3 Archive'
+    wget https://frost-release.s3-us-west-2.amazonaws.com/linux-images/artifacts/audiomini/u-boot.img
+fi
+cp u-boot.img LinuxBootImageFileGenerator/Image_partitions/Pat_1_vfat/;
+
+# Make the Image_partitions/Pat_3_raw Folder
+mkdir -p LinuxBootImageFileGenerator/Image_partitions/Pat_3_raw;
+# Check for the Audio Mini Preloader
+if [ -f audiomini_preloader.bin ]; then
+    echo 'Using the local Audio Mini Preloader.'
+else
+    echo 'Downloading the Audio Mini Preloader from S3 Archive.'
+    wget https://frost-release.s3-us-west-2.amazonaws.com/linux-images/artifacts/audiomini/audiomini_preloader.bin
+fi
+cp audiomini_preloader.bin LinuxBootImageFileGenerator/Image_partitions/Pat_3_raw;
+
+# Make the Image_partitions/Pat_2_ext3 Folder
+mkdir -p LinuxBootImageFileGenerator/Image_partitions/Pat_2_ext3;
+# Check for the Root File System
+if [ -d ubuntu-base ]; then
+    echo 'Using the local extracted file system - ubuntu-base.'
+else
+    if [ -f frost_rootfs.tar.gz ]; then
+        echo 'Using the local Root File System.'
+        tar -xhzvf frost_rootfs.tar.gz; 
```

Comment:
Does this preserve symbolic links? I guess if the final image works, then it doesn't matter.

Reply:
I don't believe it does given the `-h` is described as `follow symlinks; archive and dump the files they point to` (it does the same for hardlinks as well).

Reply:
Ah, I see. So links are just being replaced by the files. 

From my recollection, things will be messed up, and probably won't boot, if links are ignored. I recall running into issues where things like `/bin/sh` couldn't be found, leading to kernel panics. 

I believe `-p` preserves symbolic links, which would be an alternative to `-h`.